src/Datadog.Trace/Agent/Api.cs

```diff
@@ -205,8 +206,15 @@ private async Task<bool> SendTracesAsync(Span[][] traces, IApiRequest request)
                     if (response.ContentLength > 0 && Tracer.Instance.Sampler != null)
                     {
                         var responseContent = await response.ReadAsStringAsync().ConfigureAwait(false);
-                        var apiResponse = JsonConvert.DeserializeObject<ApiResponse>(responseContent);
-                        Tracer.Instance.Sampler.SetDefaultSampleRates(apiResponse?.RateByService);
+
+                        if (responseContent != _cachedResponse)
+                        {
+                            var apiResponse = JsonConvert.DeserializeObject<ApiResponse>(responseContent);
+
+                            Tracer.Instance.Sampler.SetDefaultSampleRates(apiResponse?.RateByService);
+
+                            _cachedResponse = responseContent;
```

Comment:
Can the rate returned by the agent change over time?

Reply:
Yes.

Reply:
I don't think the agent can change the rate at runtime, but a customer could change the configuration then restart the agent

Reply:
The Agent computes sample rates dynamically over time. The more traces it receives that have the same "signature" from the same service/env combination, the lower that service's sample rate becomes. The idea is that if one service sends many of traces that are very similar to each other, we can get away with sampling (keeping) fewer of them. Having a lot of duplicate traces is not useful and has a cost.

Reply:
Oh indeed. Missed it the first time, but now I see it: https://github.com/DataDog/datadog-agent/blob/master/pkg/trace/sampler/score.go

I need to make my optimization smarter then :/ I'd like to avoid the string allocation for the key, but it means I'll have to parse the keys sent by the agent. Ugh...

Reply:
I mean, it's still good, but it means that the `if (_sampleRates.Count == 0)` will only work for a while after the app has started